device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -168,6 +168,8 @@ public void open(boolean withRetry) throws IotHubClientException
     public void close()
     {
         this.deviceIO.close();
+        this.method = null;
+        this.twin = null;
```

Comment:
These maintain the twin and methods callbacks for this client and are assigned when the client first subscribes to twin/methods. Because they weren't being unassigned, they carried over between open/close attempts carrying some out-of-date callbacks

Reply:
Could we get NPEs as a result?

Reply:
No. There is a null check before these are used